File: androidApp/src/main/kotlin/com/virginactive/android/ui/timetable/DayTabs.kt
package com.virginactive.android.ui.timetable

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.selection.selectable
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.Role
import com.virginactive.android.theme.Spacing

@Composable
fun DayTabs(
    selected: DayTab,
    onSelect: (DayTab) -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .horizontalScroll(rememberScrollState())
            .padding(horizontal = Spacing.md, vertical = Spacing.sm),
        horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
    ) {
        DayTab.entries.forEach { tab ->
            DayTabPill(
                label = tab.label,
                isActive = tab == selected,
                onClick = { onSelect(tab) },
            )
        }
    }
}

@Composable
private fun DayTabPill(
    label: String,
    isActive: Boolean,
    onClick: () -> Unit,
) {
    val background: Color =
        if (isActive) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.surfaceVariant
    val content: Color =
        if (isActive) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurfaceVariant

    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(percent = 50))
            .selectable(
                selected = isActive,
                role = Role.Tab,
                onClick = onClick,
            )
            .background(background)
            .defaultMinSize(minHeight = Spacing.minTouchTarget)
            .padding(horizontal = Spacing.md, vertical = Spacing.sm),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.labelLarge,
            color = content,
        )
    }
}

File: iosApp/iosApp/iosApp/Timetable/DayTabs.swift
import SwiftUI
import Shared

struct DayTabs: View {
    let selected: DayTab
    let onSelect: (DayTab) -> Void

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: VaSpacing.sm) {
                ForEach(DayTab.allCases, id: \.self) { tab in
                    DayTabPill(
                        label: tab.label,
                        isActive: tab == selected,
                        onTap: { onSelect(tab) }
                    )
                }
            }
            .padding(.horizontal, VaSpacing.md)
            .padding(.vertical, VaSpacing.sm)
        }
    }
}

private struct DayTabPill: View {
    let label: String
    let isActive: Bool
    let onTap: () -> Void

    private var background: Color { isActive ? VaColor.vaRed : VaColor.surfaceVariant }
    private var content: Color { isActive ? VaColor.white : VaColor.onSurfaceVariant }

    var body: some View {
        Button(action: onTap) {
            Text(label)
                .font(.vaLabel)
                .foregroundStyle(content)
                .padding(.horizontal, VaSpacing.md)
                .padding(.vertical, VaSpacing.sm)
                .frame(minHeight: VaSpacing.minTouchTarget)
                .background(background)
                .clipShape(Capsule())
        }
        .buttonStyle(.plain)
    }
}
